### PR TITLE
Math gadget refactory

### PR DESCRIPTION
**Pull request by smtmfft**
Try using try_test!

### CODE REVIEW

File: zkevm-circuits/src/evm_circuit/util/math_gadget/abs_word.rs [hunk 1]
```diff
@@ -120,72 +120,65 @@ mod tests {
...
 
+    macro_rules! try_test_abs_x {
+        ($x:expr; is neg, abs eq $x_abs:expr, $expect_success:expr) => {{
+            try_test!(AbsWordGadgetContainer<Fr, true>, [$x, $x_abs], $expect_success)
+        }};
+        ($x:expr; is not neg, abs eq $x_abs:expr, $expect_success:expr) => {{
+            try_test!(AbsWordGadgetContainer<Fr, false>, [$x, $x_abs], $expect_success)
+        }};
+    }
+
```
Comment (Brechtpd) [inline zkevm-circuits/src/evm_circuit/util/math_gadget/abs_word.rs:143]: - Would just call the macro`try_test` in all files to make it as short as possible, I don't think there can be any confusion.
- The `is neg`, `abs eq` etc... is a bit strange to me, would just keep the syntax easy with normal parameters.

Something like `try_test!(Word::from(0), Word::from(0), true);` seems easy to read to me.

Comment (smtmfft) [reply zkevm-circuits/src/evm_circuit/util/math_gadget/abs_word.rs:143]: Sure, a plain macro calling is enough I think. I just tried more interesting features of rust macro, now I know why so many DSLs are created/based on rust.

File: zkevm-circuits/src/evm_circuit/util/math_gadget/mul_add_words.rs [hunk 2]
```diff
@@ -80,7 +80,7 @@ impl<F: Field> MulAddWordsGadget<F> {
         let overflow = carry_hi_expr.clone()
             + a_limbs[1].clone() * b_limbs[3].clone()
             + a_limbs[2].clone() * b_limbs[2].clone()
-            + a_limbs[3].clone() * b_limbs[2].clone()
+            + a_limbs[3].clone() * b_limbs[1].clone()
```
Comment (smtmfft) [inline zkevm-circuits/src/evm_circuit/util/math_gadget/mul_add_words.rs:83]: Finding this typo makes me feel this unit test has a little bit value :p

Comment (Brechtpd) [reply zkevm-circuits/src/evm_circuit/util/math_gadget/mul_add_words.rs:83]: Nice!

**Comment by smtmfft** [comment]
1 & 2 are fixed. 
For 3, I have thought about that, yes, it actually depends on the test container, if it has extra template params, the macro then looks like:
```rust
try_test_XXX!(
    <0, true>, // extra params
    [word, word, ...], // witness
    true, // expected result
)
```
That extra param line does not help on readability I thought, so I chose the plain text way, and then programmers can see the running container class on the spot.

**Review by Brechtpd — APPROVED** [review]

**Pull request merged**